docs/the-new-architecture/native-modules-custom-events.md

```diff
@@ -0,0 +1,412 @@
+
+## Step 5: Run Your App
+
+Now it's time to run your app.
+
+<Tabs groupId="package-manager" queryString defaultValue={constants.defaultPackageManager} values={constants.packageManagers}>
+<TabItem value="npm">
+```bash
+npm run ios
+```
+</TabItem>
+<TabItem value="yarn">
+```bash
+yarn run ios
+```
+</TabItem>
+</Tabs>
+
+<center>
+<img src="/docs/assets/turbo-native-modules-events-android.gif" width="30%" height="30%"/>
+</center>
+
+</TabItem>
+<TabItem value="ios" label="iOS">
+
+Assuming you followed the guide for iOS described in the [Native Modules guide](//docs/turbo-native-modules-introduction?platforms=ios&language=typescript#3-write-application-code-using-the-turbo-native-module), what's left to do is to plug the code that emit the events in your app.
+
+To do so, you have to:
+
+1. Open the `RCTNativeLocalStorage.h` file.
+2. Change the base class from `NSObject` to `NativeLocalStorageSpecBase`
+
+```diff title="RCTNativeLocalStorage.h"
+#import <Foundation/Foundation.h>
+#import <NativeLocalStorageSpec/NativeLocalStorageSpec.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+-@interface RCTNativeLocalStorage : NSObject <NativeLocalStorageSpec>
++@interface RCTNativeLocalStorage : NativeLocalStorageSpecBase <NativeLocalStorageSpec>
+
+@end
+
+NS_ASSUME_NONNULL_END
+```
+
+3. Open the `RCTNativeLocalStorage.mm` file.
+4. Modify it to emit the events when needed, for example:
+
+```diff title="RCTNativeLocalStorage.mm"
+ - (void)setItem:(NSString *)value key:(NSString *)key {
++   BOOL shouldEmitEvent = NO;
++   if (![self getItem:key]) {
++     shouldEmitEvent = YES;
++   }
+  [self.localStorage setObject:value forKey:key];
+
++   if (shouldEmitEvent) {
++    [self emitOnKeyAdded:@{@"key": key, @"value": value}];
++  }
+}
+```
+
+The `NativeLocalStorageSpecBase` is a base class that provides the `emitOnKeyAdded` method and its basic implementation and boilerplate. Thanks to this class, you don't have to handle all the conversion between Objective-C and JSI that is required to send the event to JS.
+
+In case of complex types, like the `KeyValuePair` we defined in the specs, Codegen will generate a generic dictionary that you can populate on the native side. It's your responsibility to make sure that the the keys you are using in the dictionary are the same properties that are defined in the spec type in JS.
+
+## Step 5: Run Your App
+
+Now it's time to run your app.
+
+<Tabs groupId="package-manager" queryString defaultValue={constants.defaultPackageManager} values={constants.packageManagers}>
+<TabItem value="npm">
+```bash
+npm run ios
+```
+</TabItem>
+<TabItem value="yarn">
+```bash
+yarn run ios
+```
```

Comment:
Those boxes that tell the user to run android/ios are making the guide longer. Can we omit them?